R/user_guide.R

```diff
@@ -281,60 +285,229 @@ render_setup_instructions <- function(path, pr) {
+  substr(text, start=1, stop=i-1)
+}
+
+#' @importFrom htmltools tags
+create_help <- function(pr) {
+  function(req, res) {
+    "!DEBUG `write_log_message(req, res, 'Generating Tableau Analytical Extension Overview')"
+    # Parse the endpoint path from header on RStudio Connect
+    content_path <- NULL
+    if (!rlang::is_null(req$HTTP_RSTUDIO_CONNECT_APP_BASE_URL)) {
+      base_url <- req$HTTP_RSTUDIO_CONNECT_APP_BASE_URL
+      rsc_root <- Sys.getenv("CONNECT_SERVER")
+      content_path <- gsub(rsc_root,
+                           "",
+                           base_url,
+                           fixed = TRUE
+      )
+
+      # If the path requested is not root (/), strip it from the content path
+      if (req$PATH_INFO != "/") {
+        content_path <- gsub(req$PATH_INFO,
+                             "",
+                             content_path,
+                             fixed = TRUE
+        )
+      }
+
+      # Ensure path starts with '/'
+      if (!stringi::stri_startswith(content_path, fixed = "/")) content_path <- paste0("/", content_path)
+    }
+    render_help(content_path, pr)
+  }
+}
+
+render_help <- function(path, pr) {
+  apiSpec <- pr$getApiSpec()
+  title <- apiSpec$info$title
+  version <- apiSpec$info$version
+  title_desc <- "title_desc not generated"
+  body_content <- "body_content not generated"
+  warnings <- warning_message()
+
+  desc <- ""
+  if (rlang::is_null(warnings)) {
+    desc <- markdown::markdownToHTML(text = strip_md_links(apiSpec$info$description),
+                                   fragment.only = TRUE)
+  }
+
+  title_desc <- htmltools::tagList(
+    tags$h1(
+      class="padded-fully title",
+      title,
+      if (!is.null(version)) paste0("(v", version, ")")
+    ),
+    tags$div(class = "api-desc",
+      tags$div(
+        class="padded-flat-top",
+        htmltools::HTML(desc)
+      )
+    )
+  )
+
+  body_content <- htmltools::tagList(
+    tags$h3(
+      class="subtitle",
+      "Introduction to plumbertableau and RStudio Connect"
+    ),
+    tags$div(
+      class = "padded-fully",
+      htmltools::HTML(markdown::markdownToHTML(
+        text = help_text()), fragment.only = TRUE
+      )
+    )
+  )
+  
+  as.character(htmltools::htmlTemplate(
+    system.file("template/index.html", package = "plumbertableau", mustWork = TRUE),
+    title_desc = title_desc,
+    body_content = body_content
+  ))
+}
+
+help_text <- function() {
+"
+The plumbertableau package makes it simple for R developers to build [Tableau Analytics Extensions](https://tableau.github.io/analytics-extensions-api/) by lightly annotating Plumber APIs and deploy those extensions to [RStudio Connect](https://www.rstudio.com/products/connect/). It also makes it easy for Tableau developers to use those extensions, by automatically generating instructions and code examples tailored for Tableau.
```

Comment:
This text is all from the old version of the vignettes, before the rewrite that landed on Friday. Should change it to reflect those. I'll take a look in a few minutes to see if there's any section I think works.

Reply:
I suggest we pull text from this vignette: https://rstudio.github.io/plumbertableau/articles/introduction.html.